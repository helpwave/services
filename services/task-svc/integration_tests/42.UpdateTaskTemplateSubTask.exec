#!/bin/bash

echo "invalid id"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "9d21dd28e8c7", "name": "Some Updated Name"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskTemplateService/UpdateTaskTemplateSubTask 2>&1 || echo ""

echo "id does not exist"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "xaa9a5e7-2a35-4203-81bb-baf4ca17a9e9", "name": "Some Updated Name"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskTemplateService/UpdateTaskTemplateSubTask 2>&1 || echo ""

echo "Existing id, no-op"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "ffee7b8b-5663-458e-a5d2-8f9e27112ee8"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskTemplateService/UpdateTaskTemplateSubTask

echo "see before"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"created_by": "22b47b87-aadf-4e81-802d-9e23489e696c"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskTemplateService/GetAllTaskTemplatesByCreator

echo "Existing id, proper update"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "9d21dd28-0b9a-40b9-abf3-71b01c18e8c7", "name": "Some Updated Name"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskTemplateService/UpdateTaskTemplateSubTask

echo "see change"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"created_by": "22b47b87-aadf-4e81-802d-9e23489e696c"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskTemplateService/GetAllTaskTemplatesByCreator

echo "undo update for consistency"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "9d21dd28-0b9a-40b9-abf3-71b01c18e8c7", "name": "Extraction of Left Pleura, Open Approach, Diagnostic"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskTemplateService/UpdateTaskTemplateSubTask
