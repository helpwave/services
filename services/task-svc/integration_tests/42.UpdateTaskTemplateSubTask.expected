invalid id
ERROR:
  Code: InvalidArgument
  Message: invalid UUID length: 12

id does not exist
{}
Existing id, no-op
{}
see before
{}
Existing id, proper update
{}
see change
{}
undo update for consistency
{}
